shared/enry.go

```diff
@@ -0,0 +1,140 @@
+// +build darwin,cgo linux,cgo
+// +build amd64
```

Comment:
It doesn't seem a big deal, we can leave it here for now and be less restrictive or move to the client in the future if needed.